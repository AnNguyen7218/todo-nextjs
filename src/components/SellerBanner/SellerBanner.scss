@import 'config.colors';
@import 'config.gutters';
@import 'config.media';

$img-heigh-sm: 52rem;
$img-heigh-xl: 80.4rem;

.img {
  width: 100%;
  vertical-align: middle;
  object-fit: cover;
  object-position: center;
  height: $img-heigh-sm;

  @include break(xl) {
    height: $img-heigh-xl;
  }
}

.banner-wrapper {
  position: relative;
  width: 100vw;
  height: $img-heigh-sm;

  @include break(xl) {
    height: $img-heigh-xl;
  }
}

.text-wrapper {
  position: absolute;
  width: 100%;
  left: 0;
  right: 0;
  top: gutter(72);
  bottom: gutter(72);

  @include break(xl) {
    top: gutter(96);
  }
}

.text-container {
  width: 100%;
  height: 100%;
}

.title {
  color: $white;
  margin-bottom: gutter(48);

  @include break(md) {
    width: 44rem;
  }

  @include break(xl) {
    margin-bottom: gutter(0);
  }
}

.subtitle {
  color: $white;
  display: none;

  @include break(xl) {
    display: block;
    margin-top: gutter(32);
    margin-bottom: gutter(32);
    width: 57rem;
  }
}
