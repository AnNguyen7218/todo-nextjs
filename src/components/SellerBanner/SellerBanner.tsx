import React from 'react';
import { MEDIA_LG_UP, MEDIA_MD_UP } from 'constants/media';
import banner from 'assets/images/seller-banner.jpg';
import banner360 from 'assets/images/seller-banner-360.jpg';
import banner720 from 'assets/images/seller-banner-720.jpg';
import { Container, Button } from 'components';
import css from './SellerBanner.scss';

const SellerBanner = () => {
  return (
    <div className={css['banner-wrapper']}>
      <picture>
        <source media={MEDIA_LG_UP} srcSet={banner} />
        <source media={MEDIA_MD_UP} srcSet={banner720} />
        <img src={banner360} alt="Banner" className={css['img']} />
      </picture>
      <div className={css['text-wrapper']}>
        <Container className={css['text-container']}>
          <h1 className={css['title']}>
            A better way to get more customers and grow your business
          </h1>
          <p className={css['subtitle']}>
            Sell your professional services to the world’s leading enterprises
            on Intrepid’s exclusive business-to-business e-commerce marketplace
          </p>
          <Button buttonType="fixed" theme="containedColor" color="white">
            Start selling
          </Button>
        </Container>
      </div>
    </div>
  );
};

export default SellerBanner;
