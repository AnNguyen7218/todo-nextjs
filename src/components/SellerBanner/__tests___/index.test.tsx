import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SellerBanner from '../SellerBanner';

afterEach(cleanup);

describe('SellerBanner Component', () => {
  test('SellerBanner should match snapshot', () => {
    const { container } = render(<SellerBanner />);
    expect(container).toMatchSnapshot();
  });
});
