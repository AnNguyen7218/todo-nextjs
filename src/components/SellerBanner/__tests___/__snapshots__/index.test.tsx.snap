// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerBanner Component SellerBanner should match snapshot 1`] = `
<div>
  <div
    class="banner-wrapper"
  >
    <picture>
      <source
        media="(min-width: 960px)"
        srcset="seller-banner.jpg"
      />
      <source
        media="(min-width: 720px)"
        srcset="seller-banner-720.jpg"
      />
      <img
        alt="Banner"
        class="img"
        src="seller-banner-360.jpg"
      />
    </picture>
    <div
      class="text-wrapper"
    >
      <div
        class="container text-container"
      >
        <h1
          class="title"
        >
          A better way to get more customers and grow your business
        </h1>
        <p
          class="subtitle"
        >
          Sell your professional services to the world’s leading enterprises on Intrepid’s exclusive business-to-business e-commerce marketplace
        </p>
        <button
          class="button button--fixed button--containedColor button--white"
          type="button"
        >
          <div
            class="button-children"
          >
            Start selling
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
